akka-docs/src/main/paradox/typed/immutable.md

```diff
@@ -0,0 +1,58 @@
+---
+project.description: Data immutability using Project Lombok
+---
+# Immutability using Lombok
+
+A preferred best practise in akka is to have immutable messages. Scala provides case class which makes it extremely easy
+to have short and clean classes for creating immutable objects, but no such facility is easily available in Java. We can make use
+of several third party libraries which help is achieving this. One good example is Lombok.
+
+Project Lombok is a java library that automatically plugs into your editor and build tools, and helps 
+get rid of much of the boilerplate code for java development.
+
+Lombok handles the following details for you. It:
+
+* modifies fields to be `private` and `final`
+* creates getters for each field
+* creates correct `equals`, `hashCode` and a human-friendly `toString`
+* creates a constructor requiring all fields.
+
+
+### Adding Lombok to a Maven project
+To add Lombok to a Maven project, declare it as a simple dependency:
+
+```xml
+<dependency>
+    <groupId>org.projectlombok</groupId>
+    <artifactId>lombok</artifactId>
+    <version>1.16.12</version>
+</dependency>
+```
+
+### Adding Lombok to an sbt projects
+For sbt, add the following line to build.sbt
+
+    val lombok = "org.projectlombok" % "lombok" % "1.18.8"
+    libraryDependencies += lombok
+    
+    
+# Using lombok    
+
+    @Value
+    public class LombokUser {
+    
+      String name;
+    
+      String email;
+    }
+    
+The example does not demonstrate other useful Lombok feature like @Builder or @Wither which will help 
+you create builder and copy methods. Be aware that Lombok is not an immutability library but a 
+code generation library which means some setups might not create immutable objects. 
+For example, Lombok’s @Data is equivalent to Lombok’s @Value but will also synthesize mutable methods. 
+Don’t use Lombok’s @Data when creating immutable classes.
+
```

Comment:
Should we also have an example of actually using a Lombok message type in an actor or stream?

Reply:
OK. Would implement these recommendations.